packages/core/src/components/ui/ProductGallery/ProductGalleryPage.tsx

```diff
@@ -4,17 +4,33 @@ import ProductGrid from 'src/components/product/ProductGrid'
 import Sentinel from 'src/sdk/search/Sentinel'
 
 import { ProductCardProps } from 'src/components/product/ProductCard'
-import { useProducts } from './usePageProducts'
+import { memo, useMemo } from 'react'
+import { usePageProductsQuery } from 'src/sdk/product/usePageProductsQuery'
 
 interface Props {
   page: number
   title: string
   productCard?: Pick<ProductCardProps, 'showDiscountBadge' | 'bordered'>
+  itemsPerPage: number
 }
 
-function GalleryPage({ page, title, productCard }: Props) {
-  const products = useProducts(page) ?? []
-  const { itemsPerPage } = useSearch()
+function ProductGalleryPage({ page, title, productCard, itemsPerPage }: Props) {
+  const {
+    state: { term, sort, selectedFacets },
+  } = useSearch()
+
+  const { currentProductsPerPage } = usePageProductsQuery({
+    page,
+    term,
+    sort,
+    selectedFacets,
+    itemsPerPage,
```

Comment:
I have a feeling these variables should not be needed when calling this hook. Would it be complex to remove the need to explicitly provide them?

Reply:
Hmm, we need these variables to provide the correct query and manage to trigger the Sort by, Filters, Search Page Features from PLP. Do you have any idea how to make these features work without passing these variables?

Reply:
I'd call this hook inside the other hook that needs it.

Reply:
when we use `usePageProductsQuery` in the in the body of Search Page or PLP we can't use the `useSearch` because it's outside the `SearchProvider` and we use the url (search params) data. That's why we need these variables when call the hook.

Reply:
Ohhh so it throws an error, got it. I think I'd put that as one more reason to separate the hook used by the PLP and Search page and the one used by GalleryPages.

Reply:
we solve this with the pair programming :)